### PR TITLE
Improve debug implementation

### PR DESCRIPTION
**Pull request by johanjanssens**
This PR fixes markup issues with the debug output ands a new debug() template method to be able to easily check if debug is enabled for templates or not.

### CODE REVIEW

File: code/libraries/joomlatools/library/template/engine/abstract.php [hunk 1]
```diff
@@ -94,6 +94,7 @@ protected function _initialize(KObjectConfig $config)
                 'json'      => 'json_encode',
                 'format'    => 'sprintf',
                 'replace'   => 'strtr',
+                'debug'     => array($this, 'isDebug'),
```
Comment (ercanozkaya) [inline code/libraries/joomlatools/library/template/engine/abstract.php]: @johanjanssens what do you think of using isDebug for the method name as well? `debug()` feels weird to me

Comment (johanjanssens) [reply code/libraries/joomlatools/library/template/engine/abstract.php]: We could do that. We are using simple function names everywhere though for template methods? Here is the list of methods we have so far: https://github.com/joomlatools/joomlatools-pages/wiki/Functions

Comment (ercanozkaya) [reply code/libraries/joomlatools/library/template/engine/abstract.php]: @johanjanssens If I first see `debug()` as a template function I would think it was a `var_dump` replacement. That is why I suggested `isDebug()`

Comment (johanjanssens) [reply code/libraries/joomlatools/library/template/engine/abstract.php]: True, var_dump() would become dump(). Will merge it as is.

**Pull request merged**